cloud/amazon/efs.py

```diff
@@ -0,0 +1,691 @@
+        required: false
+        default: None
+        aliases: []
+    id:
+        description:
+            - ID of Amazon EFS. Either name or ID required for delete.
+        required: false
+        default: None
+        aliases: []
+    tags:
+        description:
+            - |
+                List of tags of Amazon EFS. Should be defined as dictionary
+                In case of 'present' state with list of tags and existing EFS (matched by 'name'), tags of EFS will be replaced with provided data.
+        required: false
+        default: None
+        aliases: []
+    targets:
+        description:
+            - |
+                List of mounted targets. It should be a list of dictionaries, every dictionary should include next attributes:
+                    - SubnetId - Mandatory. The ID of the subnet to add the mount target in.
+                    - IpAddress - Optional. A valid IPv4 address within the address range of the specified subnet.
+                    - SecurityGroups - Optional. List of security group IDs, of the form "sg-xxxxxxxx". These must be for the same VPC as subnet specified
+                This data may be modified for existing EFS using state 'present' and new list of mount targets.
+        required: false
+        default: None
+        aliases: []
+    wait:
+        description:
+            - |
+                In case of 'present' state should wait for EFS 'available' life cycle state (of course, if current state not 'deleting' or 'deleted')
+                In case of 'absent' state should wait for EFS 'deleted' life cycle state
+        required: false
+        default: "no"
+        choices: ["yes", "no"]
+        aliases: []
+    wait_timeout:
+        description:
+            - How long the module should wait (in seconds) for desired state before returning. Zero means wait as long as necessary.
+        required: false
+        default: 0
+        aliases: []
+extends_documentation_fragment:
+    - aws
+'''
+
+EXAMPLES = '''
+# EFS provisioning
+- efs:
+    state: present
+    name: myTestEFS
+    tags:
+        Name: myTestNameTag
+        Purpose: file-storage
+    targets:
+        - SubnetId: subnet-748c5d03
+          SecurityGroups: [ "sg-1a2b3c4d" ]
+
+# Modifying EFS data
+- efs:
+    state: present
+    name: myTestEFS
+    tags:
+        Name: myAnotherTestTag
+    targets:
+        - SubnetId: subnet-7654fdca
+          SecurityGroups: [ "sg-4c5d6f7a" ]
+
+# Deleting EFS
+- efs:
+    state: absent
+    name: myTestEFS
+
+# Searching all EFS instances with tag Name = 'myTestNameTag', in subnet 'subnet-1a2b3c4d' and with security group 'sg-4d3c2b1a'
+- efs:
+    state: list
+    tags:
+        Name: myTestNameTag
+    targets:
+        - subnet-1a2b3c4d
+        - sg-4d3c2b1a
+
+'''
+
+RETURN = '''
```

Comment:
For returned values, please use [camel_dict_to_snake_dict](https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/ec2.py#L282) to match the variable style returned by other modules.